### PR TITLE
Pass transaction context in the headers

### PR DESCRIPTION
**Pull request by efirs**

### CODE REVIEW

File: api/server/v1/tx.go [hunk 1]
```diff
@@ -14,17 +14,56 @@
...
+import (
+	"context"
+
+	"github.com/grpc-ecosystem/go-grpc-middleware/util/metautils"
+	"google.golang.org/grpc"
+	"google.golang.org/protobuf/proto"
+)
+
+var (
+	HeaderPrefix = "Tigris-"
```
Comment (himank) [inline api/server/v1/tx.go:26]: should we add `header.go` in api/server/v1 then? I am thinking then we can also move the timeout header here as well?

File: server/midddleware/pprof.go [hunk 2]
```diff
@@ -21,7 +21,7 @@ import (
 	"google.golang.org/grpc"
 )
 
-func PprofUnaryServerInterceptor() grpc.UnaryServerInterceptor {
+func pprofUnaryServerInterceptor() grpc.UnaryServerInterceptor {
```
Comment (himank) [inline server/midddleware/pprof.go:24]: any reason of changing this?

Comment (efirs) [reply server/midddleware/pprof.go:24]: It's only used inside the the package, better to have only necessary scope.

Comment (efirs) [reply server/midddleware/pprof.go:24]: It is. See above:
```
-		PprofUnaryServerInterceptor(),
+		pprofUnaryServerInterceptor(),
```
It's just to signify that the function is not used outside of `middleware` package.

Comment (himank) [reply server/midddleware/pprof.go:24]: that is fine but main point is then just change the other as well that are inside this package. Mainly - `TimeoutUnaryServerInterceptor`, `TxCtxUnaryServerInterceptor`

**Review by himank — APPROVED** [review]

File: server/midddleware/pprof.go (hunk 2 above)
Comment (efirs) [reply server/midddleware/pprof.go:24]: Yes, this will be changed in the next diff.

**Pull request merged**